core/src/main/kotlin/org/evomaster/core/output/service/GraphQLTestCaseWriter.kt

```diff
@@ -1,25 +1,53 @@
 package org.evomaster.core.output.service
 
+import com.google.inject.Inject
+import org.evomaster.client.java.controller.api.dto.SutInfoDto
 import org.evomaster.core.output.Lines
 import org.evomaster.core.problem.graphql.GraphQLAction
 import org.evomaster.core.problem.graphql.GraphQLIndividual
 import org.evomaster.core.problem.graphql.GraphQLUtils
 import org.evomaster.core.problem.graphql.GraphQlCallResult
 import org.evomaster.core.problem.httpws.service.HttpWsAction
 import org.evomaster.core.problem.httpws.service.HttpWsCallResult
-import org.evomaster.core.problem.rest.param.BodyParam
+import org.evomaster.core.remote.SutProblemException
+import org.evomaster.core.remote.service.RemoteController
 import org.evomaster.core.search.Action
 import org.evomaster.core.search.ActionResult
 import org.evomaster.core.search.EvaluatedIndividual
 import org.evomaster.core.search.gene.GeneUtils
 import org.slf4j.LoggerFactory
+import javax.annotation.PostConstruct
 
 class GraphQLTestCaseWriter : HttpWsTestCaseWriter() {
 
     companion object {
         private val log = LoggerFactory.getLogger(GraphQLTestCaseWriter::class.java)
     }
 
+    @Inject(optional = true)
+    protected lateinit var rc: RemoteController
+
+    protected lateinit var infoDto: SutInfoDto
+
+    @PostConstruct
+    protected fun initialize() {
+
+        log.debug("Initializing {}", GraphQLTestCaseWriter::class.simpleName)
+        if (!config.blackBox || config.bbExperiments) {
+            rc.checkConnection()
+
+            val started = rc.startSUT()
+            if (!started) {
+                throw SutProblemException("Failed to start the system under test")
+            }
+
+            infoDto = rc.getSutInfo()
+                ?: throw SutProblemException("Failed to retrieve the info about the system under test")
+        }
+        log.debug("Done initializing {}", GraphQLTestCaseWriter::class.simpleName)
+    }
```

Comment:
we should avoid copy&paste code. plus this add another call to the backend.
i think it would be simpler to inject a reference to `HttpWsFitness` here, make its `infoDto` public, and just read that

Reply:
Done